src/controllers/scripting/javascriptplayerproxy.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include "mixer/basetrackplayer.h"
+#include "track/track.h"
+
+#define PROPERTY_IMPL_GETTER(NAMESPACE, TYPE, NAME, GETTER) \
+    TYPE NAMESPACE::GETTER() const {                        \
+        const TrackPointer pTrack = m_pCurrentTrack;        \
+        if (pTrack == nullptr) {                            \
+            return TYPE();                                  \
+        }                                                   \
+        return pTrack->GETTER();                            \
+    }
+
+class JavascriptPlayerProxy : public QObject {
+    Q_OBJECT
+    Q_PROPERTY(bool isLoaded READ isLoaded NOTIFY trackChanged)
+    Q_PROPERTY(QString artist READ getArtist NOTIFY artistChanged)
+    Q_PROPERTY(QString title READ getTitle NOTIFY titleChanged)
+    Q_PROPERTY(QString album READ getAlbum NOTIFY albumChanged)
+    Q_PROPERTY(QString albumArtist READ getAlbumArtist NOTIFY albumArtistChanged)
+    Q_PROPERTY(QString genre READ getGenre STORED false NOTIFY genreChanged)
+    Q_PROPERTY(QString composer READ getComposer NOTIFY composerChanged)
+    Q_PROPERTY(QString grouping READ getGrouping NOTIFY groupingChanged)
+    Q_PROPERTY(QString year READ getYear NOTIFY yearChanged)
+    Q_PROPERTY(QString trackNumber READ getTrackNumber NOTIFY trackNumberChanged)
+    Q_PROPERTY(QString trackTotal READ getTrackTotal NOTIFY trackTotalChanged)
```

Comment:
I'm wondering if it is necessary to implement `NOTIFY` here. Since this code won't be used by QML (as it has its own implementation), there is no need to provide this to support property binding. This will allow you to remove most of the `signals` and reduce the code duplication with `src/qml/qmltrackproxy.h`

Reply:
Yup, we should def also use signals from JS. Otherwise we'll need yet another `makeConnection` like facility or people will abuse (likely only tangentially related) COs instead.

Reply:
Ah sorry, I didn't see that this will actively be supported in the script. This is looking good then. The duplication remains a bit of a bummer IMO, but don't want to block this further so if this isn't a problem for other member, happy to ignore!

Reply:
Can I resolve this so it's clearer for me what's left?